### ISSUE TITLE
FF: Implement Chip Display Toggle for Tag and Value Inputs

### ISSUE THREAD

**Issue by zackcl** [CarnegieLearningWeb/UpGrade#1783]
## Summary
As a user, I want to see only the first two chips in the tag/value input fields and page displays by default, with an option to view all chips, so that the interface remains clean and uncluttered.

## Acceptance Criteria

**Tag/Value Input Fields in Modals:**
- When a tag/value input field in a modal contains more than two chips, display only the first two chips followed by a "+ N more" text.
- The "+ N more" text should reflect the count of additional chips hidden (e.g., "+ 3 more" if there are 5 chips in total).
- On focus, the input field should display all chips, replacing the "+ N more" text with the complete list of chips.
- When the field loses focus, revert to displaying only the first two chips with the "+ N more" text.

**Chip Display on Pages:**
- Upon page load, if there are more than two chips, display only the first two chips along with a "+ N more" text, where N is the number of additional hidden chips.
- The "+ N more" text should be underlined to indicate it is clickable.
- Clicking on "+ N more" will reveal all chips.
- The full list of chips will remain visible until the page is reloaded.

## Technical Specifications

**Frontend Changes:**
- Modify the chip rendering logic to conditionally display chips based on the total count.
- Ensure that the "+ N more" text is dynamically updated based on the number of additional chips hidden.
- Implement CSS styling to underline the "+ N more" text to indicate its clickability.

**Interactions:**
- Implement a focus event listener on the tag/value input fields to expand the chips.
- Implement a click event listener on the "+ N more" text to display all chips.

## Screenshots for Reference
Refer to the attached screenshots for visual context:

<img width="500" alt="Add Feature Flag (1)" src="https://github.com/user-attachments/assets/e84cd98b-e241-46c8-a990-b387802b83e1">

<img width="500" alt="Add Feature Flag" src="https://github.com/user-attachments/assets/8de8ff8d-d3fe-47e9-9d44-19efd48cf8d3">

<img width="500" alt="Feature Flag - Add Exclude List (1)" src="https://github.com/user-attachments/assets/f6af7b0e-9989-43ad-afaa-717fd90d3094">

<img width="500" alt="Feature Flag - Add Exclude List" src="https://github.com/user-attachments/assets/3c361dd4-d94c-43b7-8541-919ca1c58c01">

<img width="1000" alt="Feature Flags Root" src="https://github.com/user-attachments/assets/4281c48f-3d10-4dfd-80b4-903885ee3b42">

<img width="1000" alt="Feature Flags Root (1)" src="https://github.com/user-attachments/assets/fe0ef42d-99cd-4052-aa73-5c32ae9abe4d">

<img width="1000" alt="Feature Flag Details (1)" src="https://github.com/user-attachments/assets/cce3a0f5-ee03-42e8-bf3f-e98835ba35d2">

<img width="1000" alt="Feature Flag Details" src="https://github.com/user-attachments/assets/cc8d91d6-c664-48ea-85a3-4224f75d2b98">

**Comment by danoswaltCL** [comment]
was this AI generated? i wish all our stories looked like this 😎  . i put 'needs refined' on this just so it gets attention and prioritized

**Comment by zackcl** [comment]
Yes, that was written by ChatGPT since my draft sucked :sweat_smile:.

**Comment by danoswaltCL** [comment]
I'm not kidding, I talked in my MUB about things like a project-manager-bot to help us with this stuff. I don't even know what that would look like, but even just having some shared prompts we can craft as a team for storywriting seems like a no-brainer

**Comment by amurphy-cl** [comment]
For root page, we can show +N more without the ability to expand, but additional N tags will be displayed in a tooltip. When navigating into the detail page, we will show all available tags.

**Comment by danoswaltCL** [comment]
Is this really a release 6.0 item?

**Comment by danoswaltCL** [comment]
k, thought so but wanted to double check before removing the tag
